include/drivers/pcie/controller.h

```diff
@@ -0,0 +1,295 @@
+/**
+ * @file
+ *
+ * @brief Public APIs for the PCIe Controllers drivers.
+ */
+
+/*
+ * Copyright (c) 2021 BayLibre, SAS
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#ifndef ZEPHYR_INCLUDE_DRIVERS_PCIE_CONTROLLERS_H_
+#define ZEPHYR_INCLUDE_DRIVERS_PCIE_CONTROLLERS_H_
+
+#include <zephyr/types.h>
+#include <devicetree/ranges.h>
```

Comment:
Just include devicetree.h. All `devicetree/foo.h` headers are private.

Reply:
Yes, I need to rebase on #38106